### PR TITLE
Add BBE for ldap user store

### PR DESCRIPTION
**Pull request by VijithaEkanayake**
Fixes: https://github.com/ballerina-platform/ballerina-lang/issues/11220

## Purpose
$subject.

### CODE REVIEW

File: examples/service_with_ldap-user-store/service_with_ldap_user_store.client.out [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Since `admin` has `test` the invocation will succeeded.
+$ curl -kv https://localhost:9090/ldapAuth/resourceOne -H "Authorization: Basic YWRtaW46YmFsbGVyaW5h"
+Hello, World!
+
+# Since the password for the `admin` is not correct. Authentication will not be successful.
```
Comment (samgnaniah) [inline examples/service_with_ldap-user-store/service_with_ldap_user_store.client.out]: Let's make this one sentence.

File: examples/service_with_ldap-user-store/service_with_ldap_user_store.client.out [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+$ curl -kv https://localhost:9090/ldapAuth/resourceOne -H "Authorization: Basic dmlqaXRoYTpiYWxsZXJpbmE"
+Authentication failure
+
+# Since resource configuration has overridden the required scopes to denote that
```
Comment (samgnaniah) [inline examples/service_with_ldap-user-store/service_with_ldap_user_store.client.out]: Change to:

# Resource configuration has overridden the required scopes to denote that
# `test` is required to invoke `resourceTwo` resource. Hence authorization will
# not be successful.

File: examples/service_with_ldap-user-store/service_with_ldap_user_store.description [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// An authentication store provider defines an authentication scheme
+// that could be used to protect endpoints. LdapAuthStoreProvider
+// is an implementation of the AuthStoreProvider interface, which allows
+// to configure LDAP and Active Directory user stores with Ballerina.
+// A service can be secured using basic authentication and optionally, by
+// enforcing authorization. LDAP server keeps the user details such as
+// usernames, passwords for authentication, and scopes for authorization.
+// Here scopes are equivalent to LDAP groups.
+// Ballerina uses the concept of scopes for authorization. A resource
+// declared in a service can be bound to one/more scope(s). In the ldap
```
Comment (samgnaniah) [inline examples/service_with_ldap-user-store/service_with_ldap_user_store.description]: Change "ldap" to "LDAP". Apply elsewhere if applicable.

File: examples/service_with_ldap-user-store/service_with_ldap_user_store.description [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+// auth provider, scopes are mapped to users by querying the ldap server.
+// In the authorization phase, the scopes of the resource are compared against
+// the scopes mapped to the user in the ldap auth provider.
+
+// LDAP server is initialized with three users namely admin, user1 and user2.
+// Each user has a password and assigned scopes. Prior to starting the
+// Ballerina service, Make sure to download embedded ldap server from below
+// git location and start the server with provided command.
```
Comment (samgnaniah) [inline examples/service_with_ldap-user-store/service_with_ldap_user_store.description]: Add "the" between "with" and "provided command".

**Comment by VijithaEkanayake** [comment]
It was decided to a BBG instead of BBE here. will link the BBG PR once it is completed.

**Pull request closed without merging**